packages/devtools_app/lib/src/timeline/flutter/event_details.dart

```diff
@@ -2,16 +2,291 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert';
+
 import 'package:flutter/material.dart';
 
-class EventDetails extends StatelessWidget {
+import '../../flutter/common_widgets.dart';
+import '../../profiler/flutter/cpu_profiler.dart';
+import '../../ui/fake_flutter/_real_flutter.dart';
+import '../../utils.dart';
+import '../timeline_model.dart';
+
+class EventDetails extends StatefulWidget {
+  @override
+  _EventDetailsState createState() => _EventDetailsState();
+}
+
+class _EventDetailsState extends State<EventDetails> {
+  // TODO(kenz): use selected event from controller once data is hooked up.
+  TimelineEvent selectedEvent;
+
+  @override
+  void initState() {
+    super.initState();
+    selectedEvent = stubUiEvent;
+  }
+
   @override
   Widget build(BuildContext context) {
-    return Container(
-      color: Colors.black26,
-      child: const Center(
-        child: Text('TODO Event Details Section'),
-      ),
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Padding(
+          padding: const EdgeInsets.only(top: 8.0),
+          child: Text(
+            '${selectedEvent.name} - ${msText(selectedEvent.time.duration)}',
+            style: Theme.of(context).textTheme.title,
+          ),
+        ),
+        const PaddedDivider.thin(),
+        Expanded(
+          child: Container(
+            child: Stack(
+              children: [
+                if (selectedEvent.isUiEvent) CpuProfilerView(),
+                if (!selectedEvent.isUiEvent) EventSummary(selectedEvent),
+              ],
+            ),
+          ),
+        ),
+      ],
     );
   }
 }
+
+class EventSummary extends StatefulWidget {
+  const EventSummary(this.event);
+
+  final TimelineEvent event;
+
+  @override
+  _EventSummaryState createState() => _EventSummaryState();
+}
+
+class _EventSummaryState extends State<EventSummary> {
+  _EventSummaryState();
+
+  static const encoder = JsonEncoder.withIndent('    ');
+
+  List<AsyncTimelineEvent> _connectedEvents;
+
+  Map<String, dynamic> _eventArgs;
+
+  TraceEvent get firstTraceEvent => widget.event.traceEvents.first.event;
+
+  @override
+  void initState() {
+    super.initState();
+    _connectedEvents ??= widget.event.isAsyncEvent
```

Comment:
this shouldn't be a stateful widget. As far as I can tell there are no calls to change state.
Write it directly as a stateless widget instead.
@pq this could be an interesting example of a "stateless widget that could be stateless" lint.

Reply:
feel free to use a factory constructor on a stateless widget if that makes the code easier to read.